packages/foam-vscode/src/features/create-from-template.ts

```diff
@@ -218,15 +299,41 @@ function currentDirectoryFilepath(filename: string) {
   return Uri.joinPath(currentDir, filename);
 }
 
+export function determineDefaultFilepath(
+  resolvedValues: Map<string, string>,
+  templateMetadata
```

Comment:
is this an implicit `any`? in that case I am surprised it didn't raise a warning from the linter.

I also noticed in other places you are using `object` for metadata.. given that we have a specific shape for the template metadata, could it be helpful to map it onto a type/interface?

Reply:
> is this an implicit any? in that case I am surprised it didn't raise a warning from the linter.

It is. I guess I just missed the warning: 

<img width="311" alt="image" src="https://user-images.githubusercontent.com/1459385/120118760-295f4f80-c162-11eb-876d-a450fa33e9e8.png">

---

I've switched it to `Map<string, string>`... if that helps?

Reply:
I am surprised it didn't show up as a warning in the PR, I thought the integration would do that (kinda like when there is an unused import/variable)... mm, to be looked into at some point.

Given the metadata has a specific type, I would just defined that, as it also provides a good in-code documentation for what info can be found (and is allowed) as template metadata, wdyt?

Reply:
> Given the metadata has a specific type, I would just defined that, as it also provides a good in-code documentation for what info can be found (and is allowed) as template metadata, wdyt?

@riccardoferretti Sounds good to me. I won't have time to make this change until next week. If you're eager, we can ship this now, and I'll do that refactor as part of my next PR.